homeassistant/components/script/config.py

```diff
@@ -110,6 +110,8 @@ async def async_validate_config(hass, config):
     scripts = {}
     for _, p_config in config_per_platform(config, DOMAIN):
         for object_id, cfg in p_config.items():
+            if object_id in scripts:
+                raise vol.Invalid(f"Duplicate script detected with name: '{object_id}'")
```

Comment:
It feels too harsh to block all scripts from starting up because 1 contains a duplicate key. 

I also think that they didn't used to override one another but instead the 2nd one would get `_2` appended.

I would suggest we log a warning and skip the ones with duplicate keys.

Reply:
> I also think that they didn't used to override one another but instead the 2nd one would get _2 appended.

They have always overwritten each other since support for this part was added 12 months ago. It never got anything appended. Before that, you couldn't use packages with scripts in this way.

> I would suggest we log a warning and skip the ones with duplicate keys.

👍  Will adjust